src/ServerScriptService/InstanceComponents/Structures/Observer.lua

```diff
@@ -0,0 +1,36 @@
+---@diagnostic disable-next-line: undefined-global
+local Players = game:GetService("Players")
+local ReplicatedStorage = game:GetService("ReplicatedStorage")
+local RunService = game:GetService("RunService")
+
+local ObserverConfig = require(ReplicatedStorage.Common.Config.Structures.Observer)
+local Range = ObserverConfig.Range
+
+local Observer = {}
+Observer.__index = Observer
+
+function Observer.new(instance)
+	local self = setmetatable({}, Observer)
+	self.Instance = instance
+
+    RunService.Heartbeat:Connect(function()
+        if self.Instance == nil then return end
+        for _,v in pairs(Players:GetPlayers()) do
+            if not v.Character then return end
```

Comment:
The loop would stop running for all players, if the character of one is nil. This can be fixed by using `continue` instead of `return`